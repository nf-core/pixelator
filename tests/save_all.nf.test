nextflow_pipeline {

    name "Test pipeline with --save_all"
    script "../main.nf"

    test("Params: --save_all") {

        when {
            params {
                pipelines_testdata_base_path = "https://raw.githubusercontent.com/nf-core/test-datasets/pixelator/"
                input                        = "$pipelines_testdata_base_path/samplesheet/mpx/samplesheet_mpx_scsp_v1_immunology1.csv"
                input_basedir                = "$pipelines_testdata_base_path/testdata/mpx/"
                outdir                       = "$outputDir"
                save_all                     = true
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = []

            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }

    test("Params: --save_all - stub") {

        options "-stub"

        when {
            params {
                pipelines_testdata_base_path = "https://raw.githubusercontent.com/nf-core/test-datasets/pixelator/"
                input                        = "$pipelines_testdata_base_path/samplesheet/mpx/samplesheet_mpx_scsp_v1_immunology1.csv"
                input_basedir                = "$pipelines_testdata_base_path/testdata/mpx/"
                outdir = "$outputDir"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = []
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }
}
