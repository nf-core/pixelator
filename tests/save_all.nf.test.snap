{
    "Params: --save_all": {
        "content": [
            10,
            [
                "pipeline_info",
                "pipeline_info/nf_core_pixelator_software_mqc_versions.yml",
                "pipeline_info/nf_core_pixelator_software_versions.yml",
                "pixelator",
                "pixelator/adapterqc",
                "pixelator/adapterqc/sample01.failed.fastq.gz",
                "pixelator/adapterqc/sample01.meta.json",
                "pixelator/adapterqc/sample01.processed.fastq.gz",
                "pixelator/adapterqc/sample01.report.json",
                "pixelator/amplicon",
                "pixelator/amplicon/sample01.merged.fastq.gz",
                "pixelator/amplicon/sample01.meta.json",
                "pixelator/amplicon/sample01.report.json",
                "pixelator/analysis",
                "pixelator/analysis/sample01.analysis.dataset.pxl",
                "pixelator/analysis/sample01.meta.json",
                "pixelator/analysis/sample01.report.json",
                "pixelator/annotate",
                "pixelator/annotate/sample01.annotate.dataset.pxl",
                "pixelator/annotate/sample01.meta.json",
                "pixelator/annotate/sample01.report.json",
                "pixelator/collapse",
                "pixelator/collapse/sample01.collapsed.parquet",
                "pixelator/collapse/sample01.meta.json",
                "pixelator/collapse/sample01.report.json",
                "pixelator/demux",
                "pixelator/demux/sample01.failed.fastq.gz",
                "pixelator/demux/sample01.meta.json",
                "pixelator/demux/sample01.processed-AAAGCAAC.fastq.gz",
                "pixelator/demux/sample01.processed-AACAAGAC.fastq.gz",
                "pixelator/demux/sample01.processed-AACCTTCC.fastq.gz",
                "pixelator/demux/sample01.processed-AACGGCTA.fastq.gz",
                "pixelator/demux/sample01.processed-AAGCATAG.fastq.gz",
                "pixelator/demux/sample01.processed-AAGTAGCT.fastq.gz",
                "pixelator/demux/sample01.processed-AAGTCGTG.fastq.gz",
                "pixelator/demux/sample01.processed-ACATTGAC.fastq.gz",
                "pixelator/demux/sample01.processed-ACCAGTAC.fastq.gz",
                "pixelator/demux/sample01.processed-ACCTTGTG.fastq.gz",
                "pixelator/demux/sample01.processed-ACTCGGAA.fastq.gz",
                "pixelator/demux/sample01.processed-ACTCTCAC.fastq.gz",
                "pixelator/demux/sample01.processed-ACTGTGTC.fastq.gz",
                "pixelator/demux/sample01.processed-AGAGGCTC.fastq.gz",
                "pixelator/demux/sample01.processed-AGCCCGAA.fastq.gz",
                "pixelator/demux/sample01.processed-AGCTACTA.fastq.gz",
                "pixelator/demux/sample01.processed-AGGATGTT.fastq.gz",
                "pixelator/demux/sample01.processed-AGGGCAGT.fastq.gz",
                "pixelator/demux/sample01.processed-AGTTATCG.fastq.gz",
                "pixelator/demux/sample01.processed-ATAGGGGA.fastq.gz",
                "pixelator/demux/sample01.processed-ATGACTGC.fastq.gz",
                "pixelator/demux/sample01.processed-ATTCGCCT.fastq.gz",
                "pixelator/demux/sample01.processed-ATTGGCAC.fastq.gz",
                "pixelator/demux/sample01.processed-ATTTCGAG.fastq.gz",
                "pixelator/demux/sample01.processed-CAATCGGC.fastq.gz",
                "pixelator/demux/sample01.processed-CACACGGT.fastq.gz",
                "pixelator/demux/sample01.processed-CACGTTTC.fastq.gz",
                "pixelator/demux/sample01.processed-CAGATCCG.fastq.gz",
                "pixelator/demux/sample01.processed-CAGTCAGT.fastq.gz",
                "pixelator/demux/sample01.processed-CATGGGCA.fastq.gz",
                "pixelator/demux/sample01.processed-CCAGACAC.fastq.gz",
                "pixelator/demux/sample01.processed-CCAGCATG.fastq.gz",
                "pixelator/demux/sample01.processed-CCGATATC.fastq.gz",
                "pixelator/demux/sample01.processed-CCTAAGAA.fastq.gz",
                "pixelator/demux/sample01.processed-CTACGACG.fastq.gz",
                "pixelator/demux/sample01.processed-CTACTCGC.fastq.gz",
                "pixelator/demux/sample01.processed-CTCAAGAG.fastq.gz",
                "pixelator/demux/sample01.processed-CTCAGATG.fastq.gz",
                "pixelator/demux/sample01.processed-CTGACCAT.fastq.gz",
                "pixelator/demux/sample01.processed-CTGTAGGA.fastq.gz",
                "pixelator/demux/sample01.processed-CTTCTTGA.fastq.gz",
                "pixelator/demux/sample01.processed-CTTGCACC.fastq.gz",
                "pixelator/demux/sample01.processed-GAAAGTCA.fastq.gz",
                "pixelator/demux/sample01.processed-GAATCCCG.fastq.gz",
                "pixelator/demux/sample01.processed-GACCACTC.fastq.gz",
                "pixelator/demux/sample01.processed-GACTGGGA.fastq.gz",
                "pixelator/demux/sample01.processed-GAGGTTAG.fastq.gz",
                "pixelator/demux/sample01.processed-GATAGGGT.fastq.gz",
                "pixelator/demux/sample01.processed-GATTGTGC.fastq.gz",
                "pixelator/demux/sample01.processed-GCACTTAG.fastq.gz",
                "pixelator/demux/sample01.processed-GCATTCTG.fastq.gz",
                "pixelator/demux/sample01.processed-GCCGGACG.fastq.gz",
                "pixelator/demux/sample01.processed-GCGCAACT.fastq.gz",
                "pixelator/demux/sample01.processed-GCTATTGA.fastq.gz",
                "pixelator/demux/sample01.processed-GCTGAACC.fastq.gz",
                "pixelator/demux/sample01.processed-GGAGCCAA.fastq.gz",
                "pixelator/demux/sample01.processed-GGTTTACC.fastq.gz",
                "pixelator/demux/sample01.processed-GTAGGACC.fastq.gz",
                "pixelator/demux/sample01.processed-GTATCGGT.fastq.gz",
                "pixelator/demux/sample01.processed-GTCTTTGT.fastq.gz",
                "pixelator/demux/sample01.processed-GTGAGTAG.fastq.gz",
                "pixelator/demux/sample01.processed-GTGCATTC.fastq.gz",
                "pixelator/demux/sample01.processed-GTTCAGAC.fastq.gz",
                "pixelator/demux/sample01.processed-GTTGTCCG.fastq.gz",
                "pixelator/demux/sample01.processed-TACTCTTG.fastq.gz",
                "pixelator/demux/sample01.processed-TATCCCTT.fastq.gz",
                "pixelator/demux/sample01.processed-TCAACGCT.fastq.gz",
                "pixelator/demux/sample01.processed-TCAGGGTG.fastq.gz",
                "pixelator/demux/sample01.processed-TCCCAGTG.fastq.gz",
                "pixelator/demux/sample01.processed-TCCCTTGC.fastq.gz",
                "pixelator/demux/sample01.processed-TCCGTAAC.fastq.gz",
                "pixelator/demux/sample01.processed-TCGTAACG.fastq.gz",
                "pixelator/demux/sample01.processed-TCTGCTCC.fastq.gz",
                "pixelator/demux/sample01.processed-TGATAGAA.fastq.gz",
                "pixelator/demux/sample01.processed-TGCATGTC.fastq.gz",
                "pixelator/demux/sample01.processed-TGGGGCTT.fastq.gz",
                "pixelator/demux/sample01.processed-TTCTGGGT.fastq.gz",
                "pixelator/demux/sample01.processed-TTGTCCAA.fastq.gz",
                "pixelator/demux/sample01.processed-TTTCTGGT.fastq.gz",
                "pixelator/demux/sample01.processed-TTTGGAAG.fastq.gz",
                "pixelator/demux/sample01.report.json",
                "pixelator/graph",
                "pixelator/graph/sample01.discarded_edgelist.parquet",
                "pixelator/graph/sample01.edgelist.parquet",
                "pixelator/graph/sample01.meta.json",
                "pixelator/graph/sample01.report.json",
                "pixelator/layout",
                "pixelator/layout/sample01.meta.json",
                "pixelator/layout/sample01.report.json",
                "pixelator/logs",
                "pixelator/logs/sample01",
                "pixelator/logs/sample01/sample01.pixelator-adapterqc.log",
                "pixelator/logs/sample01/sample01.pixelator-amplicon.log",
                "pixelator/logs/sample01/sample01.pixelator-analysis.log",
                "pixelator/logs/sample01/sample01.pixelator-annotate.log",
                "pixelator/logs/sample01/sample01.pixelator-collapse.log",
                "pixelator/logs/sample01/sample01.pixelator-demux.log",
                "pixelator/logs/sample01/sample01.pixelator-graph.log",
                "pixelator/logs/sample01/sample01.pixelator-layout.log",
                "pixelator/logs/sample01/sample01.pixelator-preqc.log",
                "pixelator/logs/sample01/sample01.pixelator-report.log",
                "pixelator/preqc",
                "pixelator/preqc/sample01.failed.fastq.gz",
                "pixelator/preqc/sample01.meta.json",
                "pixelator/preqc/sample01.processed.fastq.gz",
                "pixelator/preqc/sample01.qc-report.html",
                "pixelator/preqc/sample01.report.json",
                "pixelator/sample01.layout.dataset.pxl",
                "pixelator/sample01.qc-report.html"
            ],
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-15T14:16:04.664176925"
    },
    "Params: --save_all - stub": {
        "content": [
            10,
            [
                "pipeline_info",
                "pipeline_info/nf_core_pixelator_software_mqc_versions.yml",
                "pipeline_info/nf_core_pixelator_software_versions.yml",
                "pixelator",
                "pixelator/adapterqc",
                "pixelator/adapterqc/sample01.meta.json",
                "pixelator/adapterqc/sample01.report.json",
                "pixelator/amplicon",
                "pixelator/amplicon/sample01.meta.json",
                "pixelator/amplicon/sample01.report.json",
                "pixelator/analysis",
                "pixelator/analysis/sample01.meta.json",
                "pixelator/analysis/sample01.report.json",
                "pixelator/annotate",
                "pixelator/annotate/sample01.meta.json",
                "pixelator/annotate/sample01.report.json",
                "pixelator/collapse",
                "pixelator/collapse/sample01.meta.json",
                "pixelator/collapse/sample01.report.json",
                "pixelator/demux",
                "pixelator/demux/sample01.meta.json",
                "pixelator/demux/sample01.report.json",
                "pixelator/graph",
                "pixelator/graph/sample01.meta.json",
                "pixelator/graph/sample01.report.json",
                "pixelator/layout",
                "pixelator/layout/sample01.meta.json",
                "pixelator/layout/sample01.report.json",
                "pixelator/logs",
                "pixelator/logs/sample01",
                "pixelator/logs/sample01/sample01.pixelator-adapterqc.log",
                "pixelator/logs/sample01/sample01.pixelator-amplicon.log",
                "pixelator/logs/sample01/sample01.pixelator-analysis.log",
                "pixelator/logs/sample01/sample01.pixelator-annotate.log",
                "pixelator/logs/sample01/sample01.pixelator-collapse.log",
                "pixelator/logs/sample01/sample01.pixelator-demux.log",
                "pixelator/logs/sample01/sample01.pixelator-graph.log",
                "pixelator/logs/sample01/sample01.pixelator-layout.log",
                "pixelator/logs/sample01/sample01.pixelator-preqc.log",
                "pixelator/logs/sample01/sample01.pixelator-report.log",
                "pixelator/preqc",
                "pixelator/preqc/sample01.meta.json",
                "pixelator/preqc/sample01.report.json",
                "pixelator/sample01.html",
                "pixelator/sample01.layout.dataset.pxl"
            ],
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-26T13:37:49.716575906"
    }
}